docs/nav2_overview.md

```diff
@@ -0,0 +1,146 @@
+
+This server contains a collection of controller plugins that command the robot to follow the provided reference path.
+Each plugin exposes a ROS 2 action. Similar to the Planning Server, users can choose from several provided controller
+plugins, or they can implement their own.
+
+They are responsible for publishing `geometry_msgs::msg::Twist` messages onto the `/cmd_vel` topic.
+
+### Dealing with different robot types
+
+The Controller Server's plugins publish `geometry_msgs::msg::Twist` messages. This works fine for holonomic robots,
+but the message type presents issues for Ackermann or similarly-constrained platforms. There are two options for
+dealing with this challenge:
+
+1. convert the `Twist` messages to `ackermann_msgs::msg::AckermannDrive` messages (or a similar message type) using an
+   adapter node
+2. abuse the `Twist` message's fields to contain the desired information
+
+Option 1. is preferable because it better documents the interface changes. Check out
+[_Is "Twist" (still) a good velocity command interface_][discourse_link] for lengthy discussion on the two
+possibilities.
+
+[discourse_link]: https://discourse.ros.org/t/is-twist-still-a-good-velocity-command-interface/13218
+
+## Behavior Server
+
+This server contains a collection of various behaviors. Navigators typically use they behaviors as recovery moves when
+path planning and tracking fails. Nav2 provides some behaviors, including waiting, backing up, or calling for help.
+Unlike the Planning and Controller Servers, the Behavior Server's plugins do not have a unified interface.
+
+## Route Server
+
+This is an experimental server that plans routes using well-structured graphs, such as road graphs, instead of
+free-space. Users provide a graph file that gets overlaid on top of the static map file.
+
+In an ideal environment, the navigator could completely bypass the Planner Server, calling the Router Server to
+generate a route through the graph and an associated dense path then feeding that to the Controller Server plugin to
+track. However, there are some scenarios that would require Nav2's free-space planning capabilities:
+
+- If the distance between the robot and the route's beginning is significant. This is an example of the last-mile
+  problem.
+- If the route has a blockage and the robot needs to maneuver around it.
+
+The navigator's general data/processing flow when using the Route Server is:
+
+1. Make an action call to the Route Server to generate a route and associated dense path through the loaded graph.
+2. (Optional) If the route's start is too far away from the robot's current pose, make an action call to the Planner  
+   Server to generate a path in free-space between the robot's current pose and the route's start.
+3. (Only if 2.) Make an action call to the Controller Server to track the reference path generated in 2.
+4. Make an action call to the Controller Server to track the reference path generated in 1.
+
+> [NOTE!]
```

Comment:
Just wanted to confirm whether this `[NOTE!]` was intentional, or whether it should be `[!NOTE]`?